vowpalwabbit/allreduce.h

```diff
@@ -47,6 +44,14 @@ typedef int socket_t;
 #include "vwvis.h"
 #include <cassert>
 
+// // mutex, condition_variable cannot be used in managed C++, tell the compiler that this is unmanaged even if included
```

Comment:
is the ifdefs desired to be commented?

Reply:
ah now that i know it works in CI I can remove it. THanks for noticing